prophet-model/src/lib.rs

```diff
@@ -180,26 +195,72 @@ impl TryFrom<&BTreeMap<String, Value>> for Field {
     fn try_from(entity: &BTreeMap<String, Value>) -> Result<Self, Self::Error> {
         let name = ressa::extract(entity, "name", Value::into_string)?;
         let ty = ressa::extract(entity, "type", Value::into_string)?;
-        Ok(Field { name, ty })
+        let is_collection = ressa::extract_primitive(entity, "is_collection", Value::into_bool)?;
+        Ok(Field {
+            name,
+            ty,
+            is_collection,
+        })
     }
 }
 
 #[derive(Debug)]
-pub enum Multiplicity {
-    // ...
+pub enum Cardinality {
+    One,
+    Many,
+}
+
+impl ToString for Cardinality {
+    fn to_string(&self) -> String {
+        use Cardinality::*;
+        match self {
+            One => "1",
+            Many => "*",
+        }
+        .to_string()
+    }
 }
 
 #[derive(Debug)]
-pub struct EntityGraph(DiGraph<Entity, Multiplicity>);
+pub struct EntityGraph(DiGraph<Entity, Cardinality>);
+
+impl EntityGraph {
+    /// Attempts to create an entity graph from a list of combined Entities
+    pub fn try_new(entities: &[Entity]) -> Option<EntityGraph> {
+        let mut graph = DiGraph::new();
+        let indices = add_nodes_inner(&mut graph, entities.iter().cloned());
+
+        // Add entity nodes to the graph
+        for entity in entities {
+            let entity_ndx = indices
+                .iter()
+                .find(|ndx| graph[**ndx].name == entity.name)?;
+
+            for field in entity.fields.iter() {
+                // Until we decide how to represent one-to-many in the ReSSA will only consider one-to-one here
```

Comment:
I'm confused by this comment. Aren't one-to-many relationships just captured by `is_collection` being  set? Or am I forgetting something?

Reply:
It seems like I had just forgotten to remove this comment since that is what we decided and is also implemented a few lines below. Will fix.

Reply:
It also looked like the cardinality direction was backwards so I changed it too